troposphere/static/js/components/images/common/SecondaryImageNavigation.react.js

```diff
@@ -27,10 +27,13 @@ define(function (require) {
           allImages = stores.ImageStore.getAll(),
           images = stores.ImageStore.fetchWhere({
             created_by__username: profile.get('username')
-          }) || [], 
-          favoritedImages = stores.ImageBookmarkStore.getBookmarkedImages() || [];
+          }) || [];
 
-      if(!images || !favoritedImages){
+      // only attempt to get bookmarks if there is a profile that might have them ...
+      var userLoggedIn = !!(profile && profile.get('selected_identity')),
+        favoritedImages =  userLoggedIn ? stores.ImageBookmarkStore.getBookmarkedImages() : [];
```

Comment:
To follow on ... because this is done in `Master` - you can guarantee that any usage of `context.profile` - _*when authenticated*_ - will have an object associated with it.